### PR TITLE
fix(@clayui/panel): make sure <button> has a type

### LINKED ISSUE #3896: ClayPanel: The page is refreshed after click on the Header, inside a Form Tag
-   [x] I have searched the [issues](https://github.com/liferay/clay/issues) of this repository and believe that this is not a duplicate.

## Is there an example you can provide via codesandbox.com?

Here: https://codesandbox.io/s/jovial-wind-bgk8k?file=/src/panel.js

## What are the steps to reproduce?
1. Click on Panel to Collapse/Close

## What is the expected result?
The page should not be reloaded inside a Form Tag

## Environment

| Tech  | Version |
| ----- | ------- |
| @clayui/panel  | v3.3.4  |
| React |  16.12.0       |

### PR DESCRIPTION
**Pull request by bryceosterhaus**
fixes #3896

### CODE REVIEW

File: packages/clay-panel/src/index.tsx [hunk 1]
```diff
@@ -127,6 +127,7 @@ const ClayPanel: React.FunctionComponent<IProps> & {
 						)}
 						onClick={() => setInternalExpanded(!internalExpanded)}
 						role="tab"
+						type="button"
```
Comment (matuzalemsteles) [inline packages/clay-panel/src/index.tsx:130]: @bryceosterhaus in that case, would we be able to use `ClayButton`?

**Review by matuzalemsteles — COMMENTED** [review]
Just left a comment! 🙂

File: packages/clay-panel/src/index.tsx (hunk 1 above)
Comment (bryceosterhaus) [reply packages/clay-panel/src/index.tsx]: Good catch. I just updated to make it a little more robust and updated a few more components.

**Review by matuzalemsteles — APPROVED** [review]
LGTM

**Pull request merged**